### DISCUSSION TITLE
Table creation column ordering

### DISCUSSION THREAD

**Discussion by tluyben** [Budibase/budibase#3097] [Ideas]
I am just recently trying some things out with Budibase and I was wondering, when I create a table in the Data view, the fields jump to the end after you edit them; I know I can order them in the Design view, but you want consistent ordering in the Data view as well. Is there no way to leave them ordered as I put them in drag/drop/change some metafield 'ordering' ?

**Comment by mjashanks** [comment]
Sorry - missed this one! There's no way to do this at the moment. I'm turning this into an idea, it has been brought up before.

**Comment by itsamemarkus** [comment]
I'm new to Budibase too. Thanks for all the work. I noticed this too, it would be nice to have the possibility to reorder columns in data view, because the right order helps immensely with auto generating new design screens.

**Comment by tluyben** [comment]
@mjashanks I thought I would give it a go: 

https://github.com/tluyben/budibase/commit/28bac41a6a605f95c2f5bce2a9fa9d0c70df0d0a 

Any hints; I would rather want to do this lower level and I don't like trusting on ES2015 inserting order for sorting; it works though and I'm learning the system. 

It works especially well when you are just inserting columns from scratch instead of editing an existing table. I hope you can give some feedback.

**Comment by shogunpurple** [comment]
Hi @tluyben - thank you for giving this a go.

Using an input to order the columns feels a little bit manual - the best UX here would be drag and drop on columns, where we store the ordering in the database. To stop the columns jumping around, we could check for any ordering, and if there isn't any, organise the rest alphabetically.

I'm going to turn this discussion into an issue, and add this information there - we will get it done as soon as we can.

**Reply by tluyben** [reply]
Thanks, I was just trying to see if I could get it to work and it works. Drag & drop would indeed be the better way to do it but that would be similar to this technically anyway. I can help if you want.

**Comment by NiekertDev** [comment]
Any work on this so far?

**Comment by lukuser** [comment]
Hello, Is this feature in the pipeline yet?

**Comment by FlyingNimbus** [comment]
I created my tables and columns (in external postgres db) via Budibase, I then did "Fetch Tables" and now the columns are all out of order. When using the "Form Block" I have to reorder everything.

Rearranging them in data view, and having these manually ordered columns in data act as default for the order in design view would be great!
